pkg/api/v1/types.go

```diff
@@ -1025,6 +1025,46 @@ type ReplicationControllerList struct {
 	Items []ReplicationController `json:"items" description:"list of replication controllers"`
 }
 
+// DaemonControllerSpec is the specification of a daemon controller.
+type DaemonControllerSpec struct {
+	// Template is the object that describes the pod that will be created.
+	// The Daemon Controller will create this pod on every node that matches
+	// the template's node selector (or on every node if no node selector is
+	// specified).
+	Template *PodTemplateSpec `json:"template,omitempty"`
+}
+
+// DaemonControllerStatus represents the current status of a daemon
+// controller.
+type DaemonControllerStatus struct {
+	// NodesRunningDaemon is the number of nodes running the Daemon.
```

Comment:
For the names, I suggest "NodeCount" and "DesiredNodeCount"?

But you really need to get this reviewed by @bgrant0607.

e.g., maybe these should be lists of matching/running nodes instead of just a count.